src/main/java/de/fraunhofer/aisec/cpg/passes/CallResolver.java

```diff
@@ -673,29 +929,77 @@ private ConstructorDeclaration createConstructorDummy(
   }
 
   private List<FunctionDeclaration> getInvocationCandidatesFromRecord(
-      RecordDeclaration record, String name, List<Type> signature) {
+      RecordDeclaration recordDeclaration, String name, CallExpression call) {
+    List<Type> signature = call.getSignature();
     Pattern namePattern =
-        Pattern.compile("(" + Pattern.quote(record.getName()) + "\\.)?" + Pattern.quote(name));
-    return record.getMethods().stream()
-        .filter(m -> namePattern.matcher(m.getName()).matches() && m.hasSignature(signature))
-        .map(FunctionDeclaration.class::cast)
-        .collect(Collectors.toList());
+        Pattern.compile(
+            "(" + Pattern.quote(recordDeclaration.getName()) + "\\.)?" + Pattern.quote(name));
+
+    if (lang instanceof CXXLanguageFrontend) {
```

Comment:
I think this may also be relevant to the Python frontend

Reply:
Yes that is possible. I already mentioned this to @konradweiss and we should refactor the CallResolver to handle the different languages. But that will be part of a different pr